10up/block-components
# Exploration: Refactor codebase to TypeScript

### Description of the Change


This PR aims to refactor the entire codebase of the `@10up/block-components` to use [TypeScript](https://www.typescriptlang.org/). The reason for using TypeScript is to provide consumers of the Block Components package with much better IDE integration and autocomplete when using any of the components/hooks from this package. Additionally, it allows us to gain more confidence in the code of this library without relying on the current PropTypes workflow. 

### How to test the Change


All the tests should continue to pass. Additionally, you can go into the `example/src/blocks/` folder and try out the better autocomplete/IDE integration this change would entail for the library users. 

### Changelog Entry

> Changed - Refactor block components to use TypeScript internally


### Credits

Props @fabiankaegy, @Antonio-Laguna, @ravinderk


### Checklist:
